### PR TITLE
Lock gamemode voters into ready

### PR DESCRIPTION
**Pull request by skull132**
Adds a config option to force gamemode voters into ready. This does the following:

- Observers can no longer vote.
- People locked into ready can no longer observe.
- People who disconnect but reconnect after voting will still be readied.
- People who disconnect up until round start after voting will be logged.
- A warning is displayed in the voting window for gamemode votes informing the player of this.

To anyone going to snark "But I can just disconnect after voting then", yes, you can. But this will cover 80% of the cases automatically. The other 20% can easily be handled by admins.

This is an alternative to #10964. It implements that one ontop of other features, basically. Do not merge both.

### CODE REVIEW

**Comment by skull132** [comment]
!review

File: code/modules/mob/abstract/new_player/login.dm [hunk 1]
```diff
@@ -20,3 +20,7 @@
 	player_list |= src
 
 	client.playtitlemusic()
+
+	if (client.prefs.locked_into_ready)
+		to_chat(src, SPAN_WARNING("Your ready stats has been restored."))
```
Comment (Arrow768) [inline code/modules/mob/abstract/new_player/login.dm:25]:
```suggestion
		to_chat(src, SPAN_WARNING("Your ready status has been restored."))
```

File: code/modules/mob/abstract/new_player/menu.dm [hunk 2]
```diff
@@ -227,10 +223,11 @@
 	if(SSticker.current_state <= GAME_STATE_PREGAME) // Make sure we don't ready up after the round has started
 		// Cannot join without a saved character, if we're on SQL saves.
 		if (config.sql_saves && !client.prefs.current_character)
-			alert(src, "You have not saved your character yet. Please do so before readying up.")
-			return
```
Comment (Arrow768) [inline code/modules/mob/abstract/new_player/menu.dm:231]: The removal of the return here seems unintentional

File: code/modules/mob/abstract/new_player/new_player.dm [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+	var/list/not_around = list()
+	for (var/ckey in preferences_datums)
+		var/datum/preferences/prefs = preferences_datums[ckey]
+		if (prefs.locked_into_ready && !prefs.client)
+			not_around += ckey
+
+		prefs.locked_into_ready = FALSE
 
+	if (not_around.len)
+		log_and_message_admins("The following ckeys were voted but disconnected before roundstart: [not_around.Join(" ")].", null, null)
```
Comment (Arrow768) [inline code/modules/mob/abstract/new_player/new_player.dm:11]:
```suggestion
		log_and_message_admins("The following ckeys have voted but disconnected before roundstart: [not_around.Join(" ")].", null, null)
```

**Review by Arrow768 — CHANGES_REQUESTED** [review]

File: code/modules/mob/abstract/new_player/menu.dm (hunk 2 above)
Comment (skull132) [reply code/modules/mob/abstract/new_player/menu.dm:231]: As discussed, it is not.

**Review by NonQueueingMatt — APPROVED** [review]
We have arrived, and it is now that we perform our charge. In fielty to the Shadow Council, our undying lords, and by the grace of the Maintainers, I declare the Tower-Defense-Voter-Annihilation PR to be approved. I hereby sign the death warrant of anyone with a trollface, and consign a million souls to playing the game. May Imperial justice account in all balance. The Maintainer Council protects.

File: vueui/src/components/view/misc/voting.vue [hunk 4]
```diff
@@ -3,6 +3,11 @@
     <template v-if="mode">
       <h2>Vote: {{question}}</h2>
       Time Left: {{reman}} s<hr>
+
+      <template v-if="extra_information">
+        <p><b>{{extra_information}}</b></p>
+      </template>
```
Comment (Karolis2011) [inline vueui/src/components/view/misc/voting.vue:9]:
```suggestion
      <p v-if="extra_information"><b>{{extra_information}}</b></p>
```

**Review by Karolis2011 — APPROVED** [review]

**Review by Arrow768 — APPROVED** [review]

**Pull request closed without merging**